Util/src/main/java/io/deephaven/util/thread/ThreadInitializationFactory.java

```diff
@@ -23,10 +24,19 @@ public interface ThreadInitializationFactory {
                     return clazz.getDeclaredConstructor().newInstance();
                 } catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException
                         | InstantiationException | IllegalAccessException e) {
+
+                    // Currently the default property file is shared between both the java client and the server. This
+                    // means that client-side usage will attempt to load the thread.initialization property intended for
+                    // the server which is not available on the class path.
+                    if (e instanceof ClassNotFoundException && type.startsWith("io.deephaven.server.")) {
+                        return null;
+                    }
```

Comment:
Let's create a ticket and link here; I think this is an "ok" workaround for now, but we should have a better / easier configuration answer for this in the future.